### ISSUE TITLE
Google OAuth - Failed to get an access token or user details

### ISSUE THREAD

**Issue by danieljurican** [picoauth/picoauth#2]
Hi.

I am having strange problem with Google OAuth. After first attempt to login with Google I get redirected to google login form. After login submit, i get error in Pico - Failed to get an access token,...
Now,... if I open any other google page, I am logged in,... So, somehow I do get logged in with Google account, but not inside Pico. 
Looks like I do get token for successful login, but Pico won't recognize it?
What am I missing? 
Thank you for your help.

**Comment by tumapav** [comment]
Hello,
this error message is common if a wrong OAuth client keys are set in the configuration. Check if the correct `clientId` and `clientSecret` of your application (retrieved from console.developers.google.com) are set in the `OAuth.yml` configuration file.

Here is an example configuration file that contains Google OAuth:
https://github.com/picoauth/picoauth-examples/blob/master/PicoAuth/OAuth.yml

Are you using `\League\OAuth2\Client\Provider\Google` as a provider?

**Comment by danieljurican** [comment]
Hi,
Yes, I am using Google provider as in example.  I reset the secret. 
Strange thing is that Google does recognize my login, because after (unsuccessful) login in PicoAuth, I can go straight to Gmail website and I am logged in (just by logging in PicoAuth).

**Comment by tumapav** [comment]
The fact that you stay signed in in Google infrastructure after an OAuth login is normal, it means the login was successful on Google's part, but PicoAuth couldn't retrieve information about the signed in account.

To prevent information disclosure, PicoAuth displays only a generic error "Failed to get an access token or user details." which doesn't provide enough information to troubleshoot the problem. However, more detailed error message (which will reveal why exactly is the login unsuccessful) is available if you enable logging.

You can do that by simply installing monolog (`composer require monolog/monolog`) and modifying the log section of your container.php like this
```
$log = new \Monolog\Logger('log1');
$log->pushHandler(new \Monolog\Handler\StreamHandler(__DIR__.'/picoAuth.log', \Monolog\Logger::DEBUG));
$log->pushHandler(new \Monolog\Handler\FirePHPHandler());
$log->pushProcessor(new \Monolog\Processor\PsrLogMessageProcessor());
$container->share('logger', $log);
```
See https://github.com/picoauth/picoauth/wiki/Set-up-logging for a more detailed instructions how to set it up.

After the log is set up, try to log in with Google and see the log file. This is for example a message that appears in picoAuth.log if clientId or clientSecret is wrong, but there might be other reasons.
```
[2018-10-25 08:27:50] log1.CRITICAL: OAuth2 IdentityProviderException: invalid_client, provider League\OAuth2\Client\Provider\Google {"e":"invalid_client","provider":"League\\OAuth2\\Client\\Provider\\Google"} []
```

**Comment by danieljurican** [comment]
Hi,
Thank you for logging! 
First I was testing Google SignIn locally and when moved to production site (still for testing purposes), I recreated the API project. So in the log I could see that: 
`log1.CRITICAL: OAuth2 IdentityProviderException: Access Not Configured. Google+ API has not been used in project 3412334535450 before or it is disabled`
I had to enable Google+ API and now it works!
Thank you!
